src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx

```diff
@@ -0,0 +1,225 @@
+          disabled={isSubmitting}
+        />
+
+        <div className="flex flex-col gap-2 sm:flex-row sm:justify-evenly sm:gap-6">
+          <div className="min-w-0 sm:w-1/2">
+            <ControlledFormListbox<EcosystemProjectFormData>
+              name="category"
+              label="Category"
+              placeholder="Select category"
+              options={categoryOptions}
+              disabled={isSubmitting}
+            />
+          </div>
+          <div className="min-w-0 sm:w-1/2">
+            <ControlledFormListbox<EcosystemProjectFormData>
+              name="topic"
+              label="Topic"
+              placeholder="Select topic"
+              options={subCategoryOptions}
+              disabled={isSubmitting}
+            />
+          </div>
+        </div>
+
+        <ControlledFormFileInput<EcosystemProjectFormData>
+          name="files"
+          label="Choose a Logo for your project"
+          accept={['.png', '.jpg', '.svg', '.webp']}
+          maxSize={MAX_FILE_SIZE_IN_BYTES}
+          disabled={isSubmitting}
+          description={
+            <>
+              For best quality, please submit a white logo with a transparent
+              background, at least 1000px by 1000px, and under 100KB. You can
+              use tools like{' '}
+              <a
+                href="https://squoosh.app/"
+                className="text-brand-300 underline hover:text-brand-400"
+                rel="noopener noreferrer"
+                target="_blank"
+                aria-label='Opens "Sqoosh" in a new tab'
+              >
+                Sqoosh
+              </a>{' '}
+              to compress your image.
+            </>
+          }
+        />
+
+        <ControlledFormInput<EcosystemProjectFormData>
+          name="websiteUrl"
+          label="Website URL"
+          placeholder="Website URL"
+          type="url"
+          disabled={isSubmitting}
+        />
+
+        <ControlledFormInput<EcosystemProjectFormData, StringOrUndefined>
+          optionalLabel
+          name="youtubeUrl"
+          label="Youtube Video URL"
+          placeholder="Video URL"
+          type="url"
+          disabled={isSubmitting}
+        />
+
+        <ControlledFormInput<EcosystemProjectFormData, StringOrUndefined>
+          optionalLabel
+          name="githubUrl"
+          label="GitHub Repository URL"
+          placeholder="GitHub Repository URL"
+          type="url"
+          disabled={isSubmitting}
+        />
+
+        <ControlledFormInput<EcosystemProjectFormData, StringOrUndefined>
+          optionalLabel
+          name="xHandle"
+          label="X Handle"
+          placeholder="X Handle"
+          type="url"
+          disabled={isSubmitting}
+        />
+      </FormSection>
+      <Button
+        variant="primary"
+        disabled={isSubmitting}
+        className="w-full sm:w-auto"
+      >
+        {isSubmitting ? 'Submitting...' : 'Submit Project'}
+      </Button>
```

Comment:
Think I included it in one of comments when I was testing the form - if we could disable the button when the form is empty?

Reply:
Ah yes, you're right. I had an impression that it reloaded the page, but I can see now the errors 👍